cg/meta/transfer/external_data.py

```diff
@@ -0,0 +1,202 @@
+        return bundle_result
+
+    def check_fastq_md5sums(self, fastq_paths: List[Path]) -> List[Path]:
+        failed_sums: List[Path] = []
+        for path in fastq_paths:
+            if Path(str(path) + ".md5").exists():
+                given_md5sum: str = extract_md5sum(md5sum_file=Path(str(path) + ".md5"))
+                if not check_md5sum(file_path=path, md5sum=given_md5sum):
+                    failed_sums.append(path)
+        return failed_sums
+
+    def add_files_to_bundles(
+        self, fastq_paths: List[Path], last_version: Version, lims_sample_id: str
+    ):
+        """Adds the given fastq files to the the hk-bundle"""
+        for path in fastq_paths:
+            LOG.info("Adding path %s to bundle %s in housekeeper" % (path, lims_sample_id))
+            self.housekeeper_api.add_file(path=path, version_obj=last_version, tags=HK_FASTQ_TAGS)
+
+    def add_transfer_to_housekeeper(self, ticket_id: int, dry_run: bool = False) -> None:
+        """Creates sample bundles in housekeeper and adds the files corresponding to the ticket to the bundle"""
+        cases: List[models.Family] = self.status_db.get_cases_from_ticket(ticket_id=ticket_id).all()
+        cust: str = self.status_db.get_customer_id_from_ticket(ticket_id=ticket_id)
+        for case in cases:
+            links = case.links
+            for link in links:
+                lims_sample_id: str = link.sample.internal_id
+                paths: List[Path] = self.get_all_paths(lims_sample_id=lims_sample_id, customer=cust)
+                if dry_run:
+                    LOG.info(
+                        "Would have added %s to housekeeper and linked associated files, but this is dry-run",
+                        lims_sample_id,
+                    )
+                    return
+                last_version: Version = self.housekeeper_api.get_create_version(
+                    bundle=lims_sample_id
+                )
+                fastq_paths_to_add: List[Path] = self.housekeeper_api.check_bundle_files(
+                    file_paths=paths,
+                    bundle_name=lims_sample_id,
+                    last_version=last_version,
+                    tags=HK_FASTQ_TAGS,
+                )
+                failed_sums: List[Path] = self.check_fastq_md5sums(fastq_paths_to_add)
+                if len(failed_sums) != 0:
+                    LOG.info(
+                        "Changes in housekeeper will not be committed and no cases will be added."
+                    )
+                    return
+                self.add_files_to_bundles(
+                    fastq_paths=fastq_paths_to_add,
+                    last_version=last_version,
+                    lims_sample_id=lims_sample_id,
+                )
+                LOG.info("The md5 files match the md5sum for sample {}".format(lims_sample_id))
+        self.housekeeper_api.commit()
+        self.status_db.set_case_action(action="analyze", case=[case.internal_id for case in cases])
```

Comment:
better to make this set action on one case, then use [map](https://www.geeksforgeeks.org/python-map-function/) to apply to a list of arguments

Reply:
Map did not want to work so I went with a for loop instead